src/VulkanSample.cpp

```diff
@@ -261,16 +262,30 @@ struct CommandLineParameters
     }
 } g_CommandLineParameters;
 
-void SetDebugUtilsObjectName(VkObjectType type, uint64_t handle, const char* name)
+void SetDebugUtilsObjectName(VkObjectType type, uint64_t handle, std::string name)
```

Comment:
Big objects like strings in C++ should be passed by `const&` not by value, because passing them by value involves additional object creation and copy, which involves dynamic memory allocation.